File: leetsync/1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int n = nums.length;
        int maxSum = nums[0], currentSum = nums[0];
        for (int i = 1; i < n; i++) {
            if (nums[i - 1] < nums[i]) {
                currentSum += nums[i];
            } else {
                currentSum = nums[i];
            }
            maxSum = Math.max(maxSum, currentSum);
        }
        return maxSum;
    }
}

File: leetsync/1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.go
func maxAscendingSum(nums []int) int {
	n := len(nums)
	maxSum, currentSum := nums[0], nums[0]
	for i := 1; i < n; i++ {
		if nums[i-1] < nums[i] {
			currentSum += nums[i]
		} else {
			currentSum = nums[i]
		}
		maxSum = maxInt(maxSum, currentSum)
	}
	return maxSum
}
func maxInt(a, b int) int {
	if a > b {
		return a
	}
	return b
}